pyadjoint/__init__.py

```diff
@@ -21,5 +21,6 @@
 from .optimization.optimization_problem import MinimizationProblem
 from .optimization.ipopt_solver import IPOPTSolver
 from .optimization.rol_solver import ROLSolver
+from .optimization.rol_solver_checkpointed import LinMoreOptimiser, ROLCheckpointManager 
```

Comment:
Maybe we can bring these in from `rol_solver.py` instead of at the top level? I'm not a huge fan of having two files with the
```python
try:
   import ROL
   ...
except:
   ...
```
pattern, could instead change to:

```suggestion
from .optimization.rol_solver import ROLSolver, LinMoreOptimiser, ROLCheckpointManager
```

And then
```python
try:
    import ROL
    from .rol_solver_checkpointed import LinMoreOptimiser, ROLCheckpointManager
    ...
```
in `rol_solver.py` (if they are indeed needed as unqualified imports; maybe you should have to opt-in to checkpointing by importing them from `pyadjoint.optimization.checkpointing` or something)